reward/stakingInfo.go

```diff
@@ -0,0 +1,152 @@
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package reward
+
+import (
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/params"
+	"math"
+	"math/big"
+	"sort"
+)
+
+const (
+	AddrNotFoundInCouncilNodes = -1
+	maxStakingLimit            = uint64(100000000000)
+	DefaultGiniCoefficient     = -1.0
+)
+
+var (
+	maxStakingLimitBigInt = big.NewInt(0).SetUint64(maxStakingLimit)
+)
+
+// StakingInfo contains staking information.
+type StakingInfo struct {
+	BlockNum uint64 // Block number where staking information of Council is fetched
+
+	// Information retrieved from AddressBook smart contract
+	CouncilNodeAddrs     []common.Address // NodeIds of Council
+	CouncilStakingdAddrs []common.Address // Address of Staking account which holds staking balance
+	CouncilRewardAddrs   []common.Address // Address of Council account which will get block reward
+	KIRAddr              common.Address   // Address of KIR contract
+	PoCAddr              common.Address   // Address of PoC contract
+
+	UseGini bool
+	Gini    float64 // gini coefficient
+
+	// Derived from CouncilStakingAddrs
+	CouncilStakingAmounts []uint64 // Staking amounts of Council
+}
+
+func newEmptyStakingInfo(blockNum uint64) (*StakingInfo, error) {
+	stakingInfo := &StakingInfo{
+		BlockNum:              blockNum,
+		CouncilNodeAddrs:      make([]common.Address, 0, 0),
+		CouncilStakingdAddrs:  make([]common.Address, 0, 0),
+		CouncilRewardAddrs:    make([]common.Address, 0, 0),
+		KIRAddr:               common.Address{},
+		PoCAddr:               common.Address{},
+		CouncilStakingAmounts: make([]uint64, 0, 0),
+		Gini:                  DefaultGiniCoefficient,
+		UseGini:               false,
+	}
+	return stakingInfo, nil
+}
+
+func newStakingInfo(bc *blockchain.BlockChain, helper governanceHelper, blockNum uint64, nodeIds []common.Address, stakingAddrs []common.Address, rewardAddrs []common.Address, KIRAddr common.Address, PoCAddr common.Address) (*StakingInfo, error) {
+	intervalBlock := bc.GetBlockByNumber(blockNum)
+	statedb, err := bc.StateAt(intervalBlock.Root())
+	if err != nil {
+		logger.Trace("Failed to make a state for interval block", "interval blockNum", blockNum, "err", err)
+		return nil, err
+	}
+
+	// Get balance of stakingAddrs
+	stakingAmounts := make([]uint64, len(stakingAddrs))
+	for i, stakingAddr := range stakingAddrs {
+		tempStakingAmount := big.NewInt(0).Div(statedb.GetBalance(stakingAddr), big.NewInt(0).SetUint64(params.KLAY))
+		if tempStakingAmount.Cmp(maxStakingLimitBigInt) > 0 {
+			tempStakingAmount.SetUint64(maxStakingLimit)
+		}
+		stakingAmounts[i] = tempStakingAmount.Uint64()
+	}
+
+	var useGini bool
+	if res, err := helper.GetItemAtNumberByIntKey(blockNum, params.UseGiniCoeff); err != nil {
+		logger.Trace("Failed to get useGiniCoeff from governance", "blockNum", blockNum, "err", err)
+		return nil, err
+	} else {
+		useGini = res.(bool)
```

Comment:
`GetItemAtNumberByIntKey` returns `uint64`. Is this ok to convert to `bool`?
How about this?
```
if res > 0 {
    useGini = true
} else {
    useGini = false
}
```